grails-app/services/org/pih/warehouse/importer/PurchaseOrderActualReadyDateImportDataService.groovy

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) 2012 Partners In Health.  All rights reserved.
+ * The use and distribution terms for this software are covered by the
+ * Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0.php)
+ * which can be found in the file epl-v10.html at the root of this distribution.
+ * By using this software in any fashion, you are agreeing to be bound by
+ * the terms of this license.
+ * You must not remove this notice, or any other, from this software.
+ **/
+package org.pih.warehouse.importer
+
+import grails.core.GrailsApplication
+import grails.gorm.transactions.Transactional
+import org.pih.warehouse.core.Constants
+import org.pih.warehouse.order.Order
+import org.pih.warehouse.order.OrderItem
+import org.pih.warehouse.order.OrderService
+import org.pih.warehouse.product.Product
+
+import java.text.ParseException
+
+@Transactional
+class PurchaseOrderActualReadyDateImportDataService implements ImportDataService {
+    GrailsApplication grailsApplication
+    OrderService orderService
+
+    @Override
+    void validateData(ImportDataCommand command) {
+        log.info "Validate data ${command.filename}"
+
+        command.data.eachWithIndex { params, index ->
+            Order order
+            OrderItem orderItem
+            Product product
+
+            // Product validation
+            if (!params["productCode"]) {
+                command.errors.reject("Row ${index + 1}: 'productCode' is required")
```

Comment:
I am wondering whether those errors should be translatable. Do you think there's a need for that? Or we can leave it as is? :thinking: